src/vivarium_gates_iv_iron/components/pregnancy.py

```diff
@@ -40,25 +45,54 @@ def setup(self, builder: Builder):
                                                      key_columns=['sex'],
                                                      parameter_columns=['age', 'year'])
         self.conception_rate = builder.value.register_rate_producer('conception_rate', source=conception_rate)
+        conception_rate_data = conception_rate_data.set_index([col for col in metadata.ARTIFACT_INDEX_COLUMNS if col != "location"])
 
         outcome_probabilities = self.load_pregnancy_outcome_probabilities(builder)
         self.outcome_probabilities = builder.lookup.build_table(outcome_probabilities,
                                                                 key_columns=['sex'],
                                                                 parameter_columns=['age', 'year'])
-        # TODO remove age and sex columns when done debugging
-        self.population_view = builder.population.get_view(self.columns_created + ['age', 'sex'])
+
+        life_expectancy_data = builder.data.load("population.theoretical_minimum_risk_life_expectancy")
+        self.life_expectancy = builder.lookup.build_table(life_expectancy_data, parameter_columns=['age'])
+
+        all_cause_mortality_data = builder.data.load("cause.all_causes.cause_specific_mortality_rate").set_index([col for col in metadata.ARTIFACT_INDEX_COLUMNS if col != "location"])
+        maternal_disorder_csmr = builder.data.load("cause.maternal_disorders.cause_specific_mortality_rate").set_index([col for col in metadata.ARTIFACT_INDEX_COLUMNS if col != "location"])
+        self.background_mortality_rate = builder.lookup.build_table((all_cause_mortality_data - maternal_disorder_csmr).reset_index(),
+                                                                    key_columns=['sex'],
+                                                                    parameter_columns=['age', 'year'])
+        maternal_disorder_incidence = builder.data.load("cause.maternal_disorders.incidence_rate").set_index([col for col in metadata.ARTIFACT_INDEX_COLUMNS if col != "location"])
+
+        self.probability_maternal_deaths = builder.lookup.build_table((maternal_disorder_csmr / conception_rate_data).reset_index(),
+                                                                      key_columns=['sex'],
+                                                                      parameter_columns=['age', 'year'])
+        self.probability_non_fatal_maternal_disorder = builder.lookup.build_table(((maternal_disorder_incidence - maternal_disorder_csmr) / conception_rate_data).reset_index(),
```

Comment:
The same applies here. I'm guessing the denominator of `maternal_disorder_incidence` is total person time, in which case this equation works, but it seems like you really want the denominator to be something like person time susceptible to maternal disorders (which might be the same thing as person time not pregnant). Might be worth asking Ali about this.

Reply:
The code is following the model, but you're right it probably is worth asking. 
https://vivarium-research.readthedocs.io/en/latest/gbd2019_models/causes/maternal_disorders/index.html#id6